src/java/htsjdk/samtools/SAMRecordQueryHashComparator.java

```diff
@@ -34,7 +36,9 @@
  *
  * @author Tim Fennell
  */
-public class SAMRecordQueryHashComparator extends SAMRecordQueryNameComparator {
+public class SAMRecordQueryHashComparator extends SAMRecordQueryNameComparator implements Serializable {
+    private static final long serialVersionUID = 1L;
+
```

Comment:
This won't serialize because `Murmur3` isn't serializable, could you make it serializable as well.  There shouldn't be any problems there.